### ISSUE TITLE
SynchronizationLockException when performing game scan

### ISSUE THREAD

**Issue by martinmine** [ProjectCeleste/Celeste_Launcher#45]
# Describe the bug

When performing a game scan, the user gets an error indicating an error occurred:
![](https://user-images.githubusercontent.com/42153930/72103495-9da91100-3329-11ea-870f-bbfb0c5398f0.png)

This issue was initially reported by @Alphamplyer 
## Steps to reproduce

Perform a game scan and it will fail. In the log, the following exception is thrown:

```
[ERR] Object synchronization method was called from an unsynchronized block of code.
System.Threading.SynchronizationLockException: Object synchronization method was called from an unsynchronized block of code.
   at ProjectCeleste.GameFiles.GameScanner.GameScannnerManager.<ScanAndRepair>d__13.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Celeste_Launcher_Gui.Windows.GameScannerWindow.<>c__DisplayClass8_0.<<Window_Loaded>b__0>d.MoveNext() in C:\Users\marti\Documents\Celeste_Launcher\Celeste_Launcher_Gui\Windows\GameScannerWindow.xaml.cs:line 71
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Celeste_Launcher_Gui.Windows.GameScannerWindow.<Window_Loaded>d__8.MoveNext() in C:\Users\marti\Documents\Celeste_Launcher\Celeste_Launcher_Gui\Windows\GameScannerWindow.xaml.cs:line 71
```

## Expected behavior

Game scan should complete without errors.

**Comment by martinmine** [comment]
This is caused by the game scanner using `Monitor.Enter` for thread synchronization. To resolve this bug we would have to get rid of the `Monitor.Enter` and replace it with either SemaphreSlim or AutoResetEvent, discussed here: https://stackoverflow.com/a/35923900

**Comment by martinmine** [comment]
Fixed in 47d8fbbc4763e8a3159dab13465773bbe4b260c1
